drivers/pinctrl/pinctrl_nrf.c

```diff
@@ -162,7 +162,15 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
 			dir = NRF_GPIO_PIN_DIR_OUTPUT;
 			input = NRF_GPIO_PIN_INPUT_DISCONNECT;
 #if NRF_GPIO_HAS_CLOCKPIN && defined(NRF_SPIM_CLOCKPIN_MOSI_NEEDED)
-			clockpin = true;
+			/* CLOCKPIN setting must not be applied to SPIM12x instances. */
+			if (DT_NODE_HAS_STATUS(DT_NODELABEL(spi120), okay) ||
+			    DT_NODE_HAS_STATUS(DT_NODELABEL(spi121), okay)) {
+				if (!NRF_SPIM_IS_320MHZ_SPIM((void *)reg)) {
+					clockpin = true;
+				}
+			} else {
+				clockpin = true;
+			}
```

Comment:
this code is a bit confusing as both branches set `clockpin = true`, can it be refactored for better clarity?